appstudio-controller/controllers/appstudio.redhat.com/promotionrun_controller.go

```diff
@@ -412,10 +420,48 @@ func locateTargetManualBinding(ctx context.Context, promotionRun appstudioshared
 		}
 	}
 
-	return appstudioshared.SnapshotEnvironmentBinding{},
-		fmt.Errorf("unable to locate binding with application '%s' and target environment '%s'",
-			promotionRun.Spec.Application, promotionRun.Spec.ManualPromotion.TargetEnvironment)
+	// Binding not found, so create it
+	components := []appstudioshared.BindingComponent{}
+	componentList := appstudioshared.ComponentList{}
+	if err := k8sClient.List(ctx, &componentList); err != nil {
+		return appstudioshared.SnapshotEnvironmentBinding{}, fmt.Errorf("unable to list components: %v", err)
+	}
+	for _, component := range componentList.Items {
+		if component.Spec.Application == promotionRun.Spec.Application {
+			components = append(components, appstudioshared.BindingComponent{
+				Name: component.Spec.ComponentName,
+			})
+		}
+	}
+	if len(components) == 0 {
+		return appstudioshared.SnapshotEnvironmentBinding{}, fmt.Errorf("unable to find components for application: %s", promotionRun.Spec.Application)
+	}
+	binding := appstudioshared.SnapshotEnvironmentBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      createBindingName(&promotionRun),
+			Namespace: promotionRun.Namespace,
+			Labels: map[string]string{
+				"appstudio.application": promotionRun.Spec.Application,
+				"appstudio.environment": promotionRun.Spec.ManualPromotion.TargetEnvironment,
+			},
+		},
+		Spec: appstudioshared.SnapshotEnvironmentBindingSpec{
+			Application: promotionRun.Spec.Application,
+			Environment: promotionRun.Spec.ManualPromotion.TargetEnvironment,
+			Snapshot:    promotionRun.Spec.Snapshot,
+			Components:  components,
+		},
+	}
+	err := k8sClient.Create(ctx, &binding)
+	if err != nil {
+		return appstudioshared.SnapshotEnvironmentBinding{}, err
+	}
+
+	return binding, nil
+}
 
+func createBindingName(promotionRun *appstudioshared.PromotionRun) string {
+	return strings.ToLower(promotionRun.Spec.Application + "-" + promotionRun.Spec.ManualPromotion.TargetEnvironment + "-generated-binding")
```

Comment:
It looks like this function will potentially have an issue with long names (names greater that >250 chars). BUT, it wasn't part of the story criteria, so I'm happy to either: a) open a new story for it for next sprint or b) handle it as part of this story/PR. It's up to you. Any preference?

This is what we did for a similar issue on https://issues.redhat.com/browse/GITOPSRVCE-183

Reply:
I've updated the function to ensure the name is never greater than 250 chars.